File: find-first-palindromic-string-in-the-array/find-first-palindromic-string-in-the-array.cpp
class Solution {
public:
    string firstPalindrome(vector<string>& words) {
        for (string word : words) {
            string_view sv (word);
            if (isPal(sv)) return static_cast<string>(sv);
        } //for
        return "";
    } //firstPalindrome

    bool isPal(string_view word) {
        int n = word.size(); 
        for (int i = 0; i < n/2; i++) {
            if (word.at(i) != word.at(n - 1 - i)) return false;
        } //for
        return true;
    } //isPal
};

File: LeetCode/2231-find-first-palindromic-string-in-the-array/find-first-palindromic-string-in-the-array.java
class Solution {
    public String firstPalindrome(String[] words) {
        for (String word : words) if (isP(word)) return word;
        return "";
    } //firstP

    private boolean isP(String s) {
        int l = 0, r = s.length() - 1;
        while (l < r) {
            if (s.charAt(l++) != s.charAt(r--)) return false;
        } //while
        return true;
    } //isP
} //Sol